src/renderer/components/ChessGround.vue

```diff
@@ -216,6 +229,42 @@ export default {
     })
   },
   methods: {
+    showPromotionModal () {
```

Comment:
This code is not very clear to me. Maybe there is a more elegant way (with vuejs?).